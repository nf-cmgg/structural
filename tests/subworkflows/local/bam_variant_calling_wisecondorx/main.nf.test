nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_WISECONDORX"
    script "subworkflows/local/bam_variant_calling_wisecondorx/main.nf"
    workflow "BAM_VARIANT_CALLING_WISECONDORX"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_wisecondorx"

    options "-stub"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = [
                    [id:"ref"],
                    file(params.wisecondorx_reference, checkIfExists:true)
                ]
                input[4] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.aberrations_bed,
                    workflow.out.bins_bed,
                    workflow.out.segments_bed,
                    workflow.out.chr_statistics,
                    workflow.out.chr_plots,
                    workflow.out.genome_plot,
                    workflow.out.vcf,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
