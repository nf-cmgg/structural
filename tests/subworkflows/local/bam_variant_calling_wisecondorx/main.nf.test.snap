{
    "homo_sapiens": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    "test_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    "test_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    "test_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    "test_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    [
                        "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "wisecondorx"
                    },
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male"
                    },
                    "test.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            4,
            0
        ],
        "timestamp": "2023-11-29T16:38:20.994392088"
    }
}